examples/common-switches.mk

```diff
@@ -202,6 +203,12 @@ ifeq ($(SNTP_CLIENT),1)
 COMMONCFLAGS                   += -DOPENTHREAD_CONFIG_SNTP_CLIENT_ENABLE=1
 endif
 
+ifeq ($(THREAD_VERSION),1.1)
+COMMONCFLAGS                   += -DOPENTHREAD_THREAD_VERSION=2
+else ifeq ($(THREAD_VERSION),1.2)
+COMMONCFLAGS                   += -DOPENTHREAD_THREAD_VERSION=3
+endif
```

Comment:
Prohibit invalid `THREAD_VERSION` in case user mistyped?

Reply:
For the other invalid version string, the build will fallback to VERSION 2
https://github.com/openthread/openthread/pull/4423/files#diff-690d2857d32c8c232c5ef201bb763373R50
@bukepo

Reply:
Better to fail building explicitly and give correct values?

Reply:
@simonlingoogle I agree, it's better to fail the building.